data/data-pipeline/README.md

```diff
@@ -400,9 +404,36 @@ Additionally, you could use a pandas type schema annotation such as [pandera](ht
 
 Alternatively, or in conjunction, you could move toward using a more strictly-typed container format for read/writes such as SQL/SQLite, and use something like [SQLModel](https://github.com/tiangolo/sqlmodel) to handle more explicit type guarantees.
 
-### ETL Unit Tests
+### Fixtures used in ETL "snapshot tests"
+
+ETLs are tested for the results of their extract, transform, and load steps by
+borrowing the concept of "snapshot testing" from the world of front-end development.
+
+Snapshots are easy to update and demonstrate the results of a series of changes to
+the code base. They are good for making sure no results have changed if you don't
+expect them to change, and they are good when you expect the results to significantly
+change in a way that would be tedious to update in traditional unit tests.
+
+However, snapshot tests are also dangerous. An unthinking developer may update the
+snapshot fixtures and unknowingly encode a bug into the supposed intended output of
+the test.
+
+In order to update the snapshot fixtures of an ETL class, follow the following steps:
+
+1. If you need to manually update the fixtures, update the "furthest upstream" source
+   that is called by `_setup_etl_instance_and_run_extract`. For instance, this may
+   involve updating a zip file that imitates the source data.
+2. Run `pytest . -rsx --update_snapshots` to update snapshots for all files, or you
+   can pass a specific file name to pytest to be more precise.
+3. Re-run pytest without the `update_snapshots` flag (e.g., `pytest . -rsx`) to
+   ensure the tests now pass.
+4. Carefully check the `git diff` for the updates to all test fixtures to make sure
```

Comment:
This should be more explicit imo. What changes would you expect, for example? I think the clearer this section is, the better. Feel free to disregard.